### PR TITLE
BB-Editor - Clone odd tact from first, even tact from second (but the second)

### LINKED ISSUE #2785: Cloning "even" tacts in BBEditor
Let's say you have a pattern in the BBEditor that takes 32 steps. (at default StepsPerTact, in 4/4)
Now you want to duplicate the steps to create a similar one with 64 steps (4 tacts).

What happens right now is that using Clone Steps twice you'll get two new copies of the first 16 steps.
In most cases, this is not the expected behaviour, imo.

I think it's better if new "even" tacts are cloned from the second tact!
(starting from the fourth of course)

This way, in the example above, you would just have to clone twice to get the third tact cloned from the first and the fourth cloned from the third, and then make your variation.

If this sounds like a reasonable feature, it would just take a little edit in src/tracks/Pattern.cpp > Pattern::cloneSteps()

### PR DESCRIPTION
**Pull request by DeRobyJ**
Fixes #2785

### CODE REVIEW

File: src/tracks/Pattern.cpp [hunk 1]
```diff
@@ -481,11 +481,19 @@ void Pattern::addSteps()
 void Pattern::cloneSteps()
 {
 	int oldLength = m_steps;
+	int newTacts = (oldLength / MidiTime::stepsPerTact()) + 1;
+	int startStep = 0; //The cloning process will normally start from 0
+	if ((newTacts%2==0)&&(newTacts>3)) {
```
Comment (Umcaruje) [inline src/tracks/Pattern.cpp]: Infix operators (=, +, -, *, /, etc.) should have a space before and after by the coding conventions. Please fix this. 
https://github.com/LMMS/lmms/wiki/Coding-conventions

**Comment by DeRobyJ** [comment]
Aw, sorry, I got used to an editor that did it automatically, will do.

**Comment by Umcaruje** [comment]
Well, I actually think the clone funcion should duplicate the whole pattern, not just a part of it.

Also when I have a 4 bar pattern, your solution just clones the first bar and the second bar all the time.

To avoid some tricky code, I'd just suggest we code the clone function to duplicate your current pattern, regardless of its size.

**Comment by DeRobyJ** [comment]
I wasn't sure about that because that would make the bb block huge in just some clicks.
But since you're the second one requesting this, I'll edit it again...
I think it would be better to open a new PR. Actually with a new branch.

I'll wait for more feedback for a while, then I'll close this.

**Comment by Umcaruje** [comment]
Superseded by #2972

**Pull request closed without merging**